### PR TITLE
feat: Enhanced retries. Renamed functions.  Bumped down retries to see how it behaves in CI.

### PR DESCRIPTION
**Pull request by ebadiere**
Bumped down retries.

**Related issue(s)**: #571 

Fixes #

### CODE REVIEW

File: utils/helpers.js [hunk 1]
```diff
@@ -267,6 +269,17 @@ const unPauseAndPoll = async(ERC20Pausable) => {
 
     return false // paused
 }
+
+const pollForResponseCode = async(contract) => {
+  for (let numberOfTries = 0; process.env.MAX_RETRY <= process.env.MAX_RETRY; numberOfTries++) {
+    const event = contract.events.filter(
+      (e) => e.event === Constants.Events.ResponseCode
+    )[0].args[0]
+    return parseInt(event._hex);
```
Comment (Nana-EC) [inline utils/helpers.js:278]: Q: How does this work?
Won't the function return the on the first iteration every time?
There's no condition to decide if to return or not so i'm not sure how the for loop is providing value.
Did you mean to check for event characteristics?

File: utils/helpers.js [hunk 2]
```diff
@@ -283,5 +296,6 @@ module.exports = {
     pollForNewERC721HollowWalletOwner,
     pollForNewSignerBalance,
     pollForNewWalletBalance,
-    unPauseAndPoll
+    unPauseAndPoll,
+    pollForEvent: pollForResponseCode
```
Comment (Nana-EC) [inline utils/helpers.js:300]: nit: pollForEvent begs the question which event.
Should we leave it as pollForResponseCode since that's what it does?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Question around the loop logic

File: utils/helpers.js (hunk 1 above)
Comment (ebadiere) [reply utils/helpers.js:278]: Push up the latest.  I am not sure what happened, but it was supposed to have a for loop, and it did on my box.  I may have missed staging the latest version.

File: utils/helpers.js (hunk 2 above)
Comment (ebadiere) [reply utils/helpers.js]: I renamed it to `pollForEvent` so that it could be more generic and reused.

File: test/safe-hts-precompile/SafeHTS.js [hunk 3]
```diff
@@ -320,10 +320,8 @@ describe('SafeHTS library Test Suite', function () {
     )
 
     const signers0AfterTokenBalance = await pollForNewBalance(erc20Mock, signer1AccountID, signers0BeforeTokenBalance)
-  
-    const signers1AfterTokenBalance = parseInt(
-      await erc20Mock.balanceOf(signer2AccountID)
-    )
+
+    const signers1AfterTokenBalance = await pollForNewBalance(erc20Mock, signer2AccountID, signers1BeforeTokenBalance)
```
Comment (Nana-EC) [inline test/safe-hts-precompile/SafeHTS.js:324]: `signers1AfterTokenBalance` is never used. Wha was it added for?

File: utils/helpers.js [hunk 4]
```diff
@@ -267,6 +269,19 @@ const unPauseAndPoll = async(ERC20Pausable) => {
 
     return false // paused
 }
+
+const pollForEvent = async(contract) => {
```
Comment (Nana-EC) [inline utils/helpers.js:273]:
```suggestion
const pollForLastEvent = async(contract) => {
```

Comment (Nana-EC) [inline utils/helpers.js:273]: It's unclear was `contract` is, please add a type definition.
In https://github.com/hashgraph/hedera-smart-contracts/pull/584/files#diff-ab063aef47051b3442202d7087747109a96c12399fd2dd7c0c92a48f2c4c32a2R271 you pass in a receipt

Comment (Nana-EC) [inline utils/helpers.js:273]: https://github.com/hashgraph/hedera-smart-contracts/pull/584/files#diff-ab063aef47051b3442202d7087747109a96c12399fd2dd7c0c92a48f2c4c32a2R271 passes 2 params.
I prefer that so we should utilize that param here

**Review by Nana-EC — COMMENTED** [review]
Some improvement suggestions

File: test/safe-hts-precompile/SafeHTS.js (hunk 3 above)
Comment (ebadiere) [reply test/safe-hts-precompile/SafeHTS.js:324]: `signers1AfterTokenBalance` is used on line 344 in an `expect` validation:
`    expect(signers1AfterTokenBalance).to.equal(
      signers1BeforeTokenBalance + transferredAmount
    )
`

File: utils/helpers.js (hunk 4 above)
Comment (ebadiere) [reply utils/helpers.js]: Done.

Comment (ebadiere) [reply utils/helpers.js]: Yes, renamed to the more appropriate `transactionReceipt`.  It's a JS project so I can't declare an explicit object type.

File: test/safe-hts-precompile/SafeHTS.js (hunk 3 above)
Comment (Nana-EC) [reply test/safe-hts-precompile/SafeHTS.js:324]: Read the git change as a new variable. Seems it existed and was used already

File: utils/helpers.js [hunk 5]
```diff
@@ -267,6 +269,19 @@ const unPauseAndPoll = async(ERC20Pausable) => {
 
     return false // paused
 }
+
+const pollForLastEvent = async(transactionReceipt) => {
+  for (let numberOfTries = 0; process.env.MAX_RETRY <= process.env.MAX_RETRY; numberOfTries++) {
+    const event = transactionReceipt.events.filter(
+      (e) => e.event === Constants.Events.ResponseCode
```
Comment (Nana-EC) [inline utils/helpers.js:276]: You still have this hard coded, you should update the method signature to take in a 2nd param and use it

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Almost there.
Update `pollForLastEvent()`

File: utils/helpers.js (hunk 5 above)
Comment (ebadiere) [reply utils/helpers.js]: Updated.

**Review by Nana-EC — APPROVED** [review]

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Please update PR title and description to match changes

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**